apps/election-manager/src/utils/filenames.ts

```diff
@@ -7,26 +8,86 @@ const SUBSECTION_SEPARATOR = '_'
 
-function sanitizeString(input: string): string {
+export const SCANNER_RESULTS_FOLDER = 'cast-vote-records'
+export const BALLOT_PACKAGES_FOLDER = 'ballot-packages'
+
+export type CVRFileData = {
+  machineId: string
+  numberOfBallots: number
+  isTestModeResults: boolean
+  timestamp: Date
+}
+
+function sanitizeString(input: string, replacement = ''): string {
   return input
-    .replace(/[^a-z0-9]+/gi, WORD_SEPARATOR)
+    .replace(/[^a-z0-9]+/gi, replacement)
     .replace(/(^-|-$)+/g, '')
     .toLocaleLowerCase()
 }
 
-export const BALLOT_PACKAGES_FOLDER = 'ballot-packages'
-
 export function generateFilenameForBallotExportPackage(
   electionDefinition: ElectionDefinition,
   time: Date = new Date()
 ): string {
   const { election, electionHash } = electionDefinition
-  const electionCountyName = sanitizeString(election.county.name)
-  const electionTitle = sanitizeString(election.title)
+  const electionCountyName = sanitizeString(
+    election.county.name,
+    WORD_SEPARATOR
+  )
+  const electionTitle = sanitizeString(election.title, WORD_SEPARATOR)
   const electionInformation = `${electionCountyName}${SUBSECTION_SEPARATOR}${electionTitle}${SUBSECTION_SEPARATOR}${electionHash.slice(
     0,
     10
   )}`
   const timeInformation = moment(time).format(TIME_FORMAT_STRING)
   return `${electionInformation}${SECTION_SEPARATOR}${timeInformation}.zip`
 }
+
+export function generateElectionBasedSubfolderName(
+  election: Election,
+  electionHash: string
+): string {
+  const electionCountyName = sanitizeString(
+    election.county.name,
+    WORD_SEPARATOR
+  )
+  const electionTitle = sanitizeString(election.title, WORD_SEPARATOR)
+  return `${`${electionCountyName}${SUBSECTION_SEPARATOR}${electionTitle}`.toLocaleLowerCase()}${SUBSECTION_SEPARATOR}${electionHash.slice(
+    0,
+    10
+  )}`
+}
+
+export function parseCVRFileInfoFromFilename(
+  filename: string
+): CVRFileData | undefined {
+  const segments = filename.split(SECTION_SEPARATOR)
+  const isTestModeResults = segments.length === 4 && segments[0] === 'TEST'
+
+  const postTestPrefixSegments = isTestModeResults
+    ? segments.slice(1)
+    : segments
+  if (postTestPrefixSegments.length !== 3) {
+    return
+  }
+
+  const machineSegments = postTestPrefixSegments[0].split(SUBSECTION_SEPARATOR)
+  if (machineSegments.length !== 2 || machineSegments[0] !== 'machine') {
+    return
+  }
+  const machineId = machineSegments[1]
+
+  const ballotSegments = postTestPrefixSegments[1].split(SUBSECTION_SEPARATOR)
+  if (ballotSegments.length !== 2 || ballotSegments[1] !== 'ballots') {
+    return
+  }
+  const numberOfBallots = Number(ballotSegments[0])
+
+  const parsedTime = moment(postTestPrefixSegments[2], TIME_FORMAT_STRING)
+  return {
+    machineId,
+    numberOfBallots,
+    isTestModeResults,
+    timestamp: parsedTime.toDate(),
+  }
+}
```

Comment:
Another approach, not necessarily better, but I felt compelled to try something:

```suggestion
const CVR_FILE_PATTERN = /^(TEST__)?machine_([^_]+?)__(\d+?)_ballots__(\d{4})-(\d{2})-(\d{2})(?:_(\d{2})-(\d{2})-(\d{2}))?\.jsonl$/

export function parseCVRFileInfoFromFilename(
  filename: string
): CVRFileData | undefined {
  const match = filename.match(CVR_FILE_PATTERN)

  if (!match) {
    return
  }

  const [
    ,
    test,
    machineId,
    numberOfBallots,
    year,
    month,
    day,
    hour,
    minute,
    second,
  ] = match
  return {
    isTestModeResults: !!test,
    machineId,
    numberOfBallots: Number(numberOfBallots),
    timestamp:
      hour === undefined
        ? new Date(Number(year), Number(month) - 1, Number(day))
        : new Date(
            Number(year),
            Number(month) - 1,
            Number(day),
            Number(hour),
            Number(minute),
            Number(second)
          ),
  }
}
```